test/absinthe/phase/execution/non_null_test.exs

```diff
@@ -263,10 +277,30 @@ defmodule Absinthe.Phase.Document.Execution.NonNullTest do
       assert {:ok, %{data: data, errors: errors}} == Absinthe.run(doc, Schema)
     end
 
+    test "list of non null works when root is null" do
+      doc = """
+      {
+        nonNullListOfNullable(makeNull: true) { __typename }
+      }
+      """
+
+      data = nil
+
+      errors = [
+        %{
+          locations: [%{column: 3, line: 2}],
+          message: "Cannot return null for non-nullable field",
+          path: ["nonNullListOfNullable"]
+        }
+      ]
+
+      assert {:ok, %{data: data, errors: errors}} == Absinthe.run(doc, Schema)
```

Comment:
This test covers the case where an error is returned, but there is still no test case that shows that there is NOT an error for `[nil]` when it's allowed. I'll add one.